driver.go

```diff
@@ -357,89 +353,118 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
-		*entryPoint = driverConfig.Entrypoint
-	}
-
-	var workingDir *string // nil -> image default workingDir
-	if driverConfig.WorkingDir != "" {
-		*workingDir = driverConfig.WorkingDir
+		createOpts.ContainerBasicConfig.Entrypoint = append(createOpts.ContainerBasicConfig.Entrypoint, driverConfig.Entrypoint)
 	}
 
 	containerName := BuildContainerName(cfg)
-	memoryLimit := fmt.Sprintf("%dm", cfg.Resources.NomadResources.Memory.MemoryMB)
-	cpuShares := cfg.Resources.LinuxResources.CPUShares
-	logOpts := []string{
-		fmt.Sprintf("path=%s", cfg.StdoutPath),
-	}
 
 	// ensure to include port_map into tasks environment map
 	cfg.Env = taskenv.SetPortMapEnvs(cfg.Env, driverConfig.PortMap)
 
-	// convert environment map into a k=v list
-	allEnv := cfg.EnvList()
-
-	allVolumes, err := d.containerBinds(cfg, &driverConfig)
+	// -------------------------------------------------------------------------------------------
+	// BASIC
+	// -------------------------------------------------------------------------------------------
+	createOpts.ContainerBasicConfig.Name = containerName
+	createOpts.ContainerBasicConfig.Command = allArgs
+	createOpts.ContainerBasicConfig.Env = cfg.Env
+	createOpts.ContainerBasicConfig.Hostname = driverConfig.Hostname
+
+	createOpts.ContainerBasicConfig.LogConfiguration.Path = cfg.StdoutPath
+
+	// -------------------------------------------------------------------------------------------
+	// STORAGE
+	// -------------------------------------------------------------------------------------------
+	createOpts.ContainerStorageConfig.Init = driverConfig.Init
+	createOpts.ContainerStorageConfig.Image = driverConfig.Image
+	createOpts.ContainerStorageConfig.InitPath = driverConfig.InitPath
+	createOpts.ContainerStorageConfig.WorkDir = driverConfig.WorkingDir
+	allMounts, err := d.containerMounts(cfg, &driverConfig)
 	if err != nil {
 		return nil, nil, err
 	}
-	d.logger.Debug("binding volumes", "volumes", allVolumes)
+	createOpts.ContainerStorageConfig.Mounts = allMounts
 
-	swap := memoryLimit
-	if driverConfig.MemorySwap != "" {
-		swap = driverConfig.MemorySwap
+	// -------------------------------------------------------------------------------------------
+	// RESOURCES
+	// -------------------------------------------------------------------------------------------
+	createOpts.ContainerResourceConfig.ResourceLimits = &spec.LinuxResources{
+		Memory: &spec.LinuxMemory{},
+		CPU:    &spec.LinuxCPU{},
+	}
+	if driverConfig.MemoryReservation != "" {
+		reservation, err := memoryInBytes(driverConfig.MemoryReservation)
+		if err != nil {
+			return nil, nil, err
+		}
+		createOpts.ContainerResourceConfig.ResourceLimits.Memory.Reservation = &reservation
 	}
 
-	procFilesystems, err := getProcFilesystems()
-	swappiness := new(int64)
-	if err == nil {
-		cgroupv2 := false
-		for _, l := range procFilesystems {
-			cgroupv2 = cgroupv2 || strings.HasSuffix(l, "cgroup2")
+	if cfg.Resources.NomadResources.Memory.MemoryMB > 0 {
```

Comment:
should this be memorybytes? the limit seems off if its memoryMb * 1024 * 1024 :thinking:

Reply:
what is this limit? MemoryMB * 1024 * 1024 seems high, unless memoryMB its bytes?